js/ui/grid_core/ui.grid_core.virtual_scrolling.js

```diff
@@ -1128,17 +1128,24 @@ export const virtualScrollingModule = {
                                 if(isNextGroupItem || isNextDataItem) {
                                     wasCountableItem = true;
                                 }
-
+                                if(item.isNewRow) {
+                                    newRows.push(item);
+                                } else {
+                                    newRows.forEach(it => it.loadIndex = currentIndex);
+                                    newRows = [];
+                                    item.loadIndex = currentIndex;
+                                }
                                 item.loadIndex = currentIndex;
```

Comment:
I think we can remove the setting of the ```loadIndex``` value here
It makes that going on here a little more obvious :)

```suggestion
```